src/delve.ts

```diff
@@ -0,0 +1,154 @@
+import * as ROT from 'rot-js';
+import * as Game from './game';
+import * as World from './world';
+import { filterInPlace, randomChoices } from './utils';
+
+export function SimulateDelve(game: Game.GameState): Game.GameState {
+  //Simulate an adventure, taking the "real work" out of playing roguelikes
+  const delveSummaryMessage: string[] = [];
+
+  //leave behind any non-party members
+  const party = [...game.party];
+  filterInPlace(party, (p) => p.relationship === 'Party Member');
+
+  //Loot Simulation
+  const inventory = game.inventory;
+  const someoneWearingRingOfSearching =
+    party.filter((c) => c.ringFinger?.name === 'ring of searching').length > 0;
+  const newLootCount =
+    2 *
+    (1 + (someoneWearingRingOfSearching ? 1 : 0)) *
+    game.delveSimulation.lootMultiplier;
+  inventory.push(...randomChoices(World.LootList, newLootCount));
+  if (newLootCount) {
+    delveSummaryMessage.push('We found ' + newLootCount + ' pieces of loot.');
+  }
+
+  //Combat Simulation
+  let totalCombatIncidence = 0;
+  const simulateCombat = () => {
+    //TODO: assemble total combat power, instead of per-party member
+    party.forEach((c: World.Character) => {
+      const wearingRingOfConflict = c.ringFinger?.name === 'ring of conflict';
```

Comment:
I'd be SOOOOOOO tempted to pull all the `ring of {x}` into an `enum` or an Object hash. 😁